h5py/_hl/dataset.py

```diff
@@ -552,7 +552,14 @@ def __getitem__(self, args):
         # Perform the dataspace selection.
         selection = sel.select(self.shape, args, dsid=self.id)
 
-        if selection.nselect == 0:
+        # If we are running in MPI mode we need to check if we are set in
+        # collective IO mode to ensure we actually perform all IO modes
```

Comment:
I think the comment could use a bit of extra explanation, something like:

> ...so that it doesn't get stuck waiting for an I/O operation that some workers have skipped

(IDK if 'workers' is the standard term in MPI, but I trust you'll know this)

Reply:
Thanks. Worker would make sense, but I think "MPI process" is most clear, so I'll use that.